File: python/pages/login_page.py
from playwright.sync_api import Page


class LoginPage:
	"""Page Object Model for the login page (synchronous Playwright).

	Use `goto()` to navigate to the login page and `login()` to perform a
	simple username/password submit. Tests should use this class to keep
	selectors and navigation in one place.
	"""

	def __init__(self, page: Page):
		self.page = page

	def goto(self) -> None:
		self.page.goto("/login")

	def login(self, username: str, password: str) -> None:
		self.page.fill("#username", username)
		self.page.fill("#password", password)
		self.page.click('button[type="submit"]')

File: java/src/test/java/pages/LoginPage.java
package pages;

import com.microsoft.playwright.Page;

/**
 * Page Object Model for the login page.
 * Use goto() to navigate to the login page and login() to perform
 * a simple username/password submit. Tests should rely on this POM rather
 * than using raw Playwright calls to keep behavior centralized.
 */
public class LoginPage {
  private final Page page;

  public LoginPage(Page page) {
    this.page = page;
  }

  public void navigateTo() {
    page.navigate("https://practice.expandtesting.com/login");
  }

  /**
   * Fill username and password fields and submit the form.
   *
   * @param username Login username
   * @param password Login password
   */
  public void login(String username, String password) {
    page.fill("#username", username);
    page.fill("#password", password);
    page.click("button[type=\"submit\"]");
  }
}
